### PR TITLE
Update `metrics` properties 

### LINKED ISSUE #5251: Update docs on `metrics` preview feature
(Internal conversation: https://prisma-company.slack.com/archives/C5Z9TH6N9/p1693499282829079)

### PR DESCRIPTION
**Pull request by ruheni**
Update property names for metrics

Closes #5251

TODO:

- [x] Wait for `dev` release to update the `metrics` properties

### CODE REVIEW

File: content/200-concepts/100-components/02-prisma-client/240-metrics.mdx [hunk 1]
```diff
@@ -29,22 +29,20 @@ Prisma Client provides the following metrics:
...
   - `prisma_pool_connections_idle`: The number of pool connections that are not currently being used. These pool connections are waiting for the next datasource query to run.
-  - `prisma_client_queries_active`: The number of currently active transactions.
-  - `prisma_client_queries_wait`: The number of Prisma Client queries currently waiting for a connection because all connections are in use.
+  - `prisma_pool_connections_opened_total`: The number of currently open pool connections.
 
 - Histograms (metrics data divided into a collection of values; we call each container in the collection a "bucket"):
 
   - `prisma_client_queries_wait_histogram_ms`: The time waiting for a pool connection for all Prisma Client queries in ms.
-  - `prisma_client_queries_duration_histogram_ms`: The execution time for all executed Prisma Client queries in ms. This includes the time taken to execute all database queries, and to carry out all database engine activities, such as joining data and transforming data to the correct format.
-  - `prisma_datasource_queries_duration_histogram_ms`: The execution time for all executed Datasource queries in ms.
```
Comment (janpio) [inline content/200-concepts/100-components/02-prisma-client/240-metrics.mdx]: Why remove these?

File: content/200-concepts/100-components/02-prisma-client/240-metrics.mdx [hunk 2]
```diff
@@ -29,22 +29,20 @@ Prisma Client provides the following metrics:
   - `prisma_client_queries_total`: The total number of Prisma Client queries executed.
   - `prisma_datasource_queries_total`: The total number of datasource queries executed (SQL queries in relational databases, and commands in MongoDB).
     - The value returned by `prisma_datasource_queries_total` can be greater than `prisma_client_queries_total`, because some Prisma Client operations create multiple queries.
-  - `prisma_pool_connections_opened_total`: The total number of [pool connections](/concepts/components/prisma-client/working-with-prismaclient/connection-pool#default-connection-pool-size) opened.
+  - `prisma_pool_connections_open`: The total number of [pool connections](/concepts/components/prisma-client/working-with-prismaclient/connection-pool#default-connection-pool-size) opened.
```
Comment (janpio) [inline content/200-concepts/100-components/02-prisma-client/240-metrics.mdx]: I don't think that this counter was renamed.

File: content/200-concepts/100-components/02-prisma-client/240-metrics.mdx [hunk 3]
```diff
@@ -29,22 +29,20 @@ Prisma Client provides the following metrics:
...
 - Gauges (can increase or decrease):
 
-  - `prisma_pool_connections_open`: The number of currently open pool connections.
+  - `prisma_client_queries_active`: The number of currently active Prisma Client queries.
+  - `prisma_client_queries_wait`: The number of Prisma Client queries currently waiting for a connection because all connections are in use.
   - `prisma_pool_connections_busy`: The number of currently busy pool connections. These pool connections are currently executing a datasource query.
   - `prisma_pool_connections_idle`: The number of pool connections that are not currently being used. These pool connections are waiting for the next datasource query to run.
-  - `prisma_client_queries_active`: The number of currently active transactions.
-  - `prisma_client_queries_wait`: The number of Prisma Client queries currently waiting for a connection because all connections are in use.
+  - `prisma_pool_connections_opened_total`: The number of currently open pool connections.
```
Comment (janpio) [inline content/200-concepts/100-components/02-prisma-client/240-metrics.mdx]: This is still a counter (indicated by the `_total`).

File: content/200-concepts/100-components/02-prisma-client/240-metrics.mdx (hunk 1 above)
Comment (ruheni) [reply content/200-concepts/100-components/02-prisma-client/240-metrics.mdx]: These two were not included as part of the output. I'll add them back in.

File: content/200-concepts/100-components/02-prisma-client/240-metrics.mdx (hunk 3 above)
Comment (ruheni) [reply content/200-concepts/100-components/02-prisma-client/240-metrics.mdx]: This was included in the gauge, based on the output I got. Perhaps it's a bug? I'll retry it later and get back to you on this.

**Review by jharrell — APPROVED** [review]

**Review by janpio — COMMENTED** [review]
Can you also update the Json and Prometheus output?

File: content/200-concepts/100-components/02-prisma-client/240-metrics.mdx [hunk 4]
```diff
@@ -29,16 +29,16 @@ Prisma Client provides the following metrics:
...
 - Gauges (can increase or decrease):
 
-  - `prisma_pool_connections_open`: The number of currently open pool connections.
+  - `prisma_client_queries_active`: The number of currently active Prisma Client queries.
+  - `prisma_client_queries_wait`: The number of Prisma Client queries currently waiting for a connection because all connections are in use.
   - `prisma_pool_connections_busy`: The number of currently busy pool connections. These pool connections are currently executing a datasource query.
   - `prisma_pool_connections_idle`: The number of pool connections that are not currently being used. These pool connections are waiting for the next datasource query to run.
-  - `prisma_client_queries_active`: The number of currently active transactions.
-  - `prisma_client_queries_wait`: The number of Prisma Client queries currently waiting for a connection because all connections are in use.
+  - `prisma_pool_connections_open`: The total number of [pool connections](/concepts/components/prisma-client/working-with-prismaclient/connection-pool#default-connection-pool-size) opened.
```
Comment (janpio) [inline content/200-concepts/100-components/02-prisma-client/240-metrics.mdx]: This is not the "total number ... opened".

File: content/200-concepts/100-components/02-prisma-client/240-metrics.mdx [hunk 5]
```diff
@@ -38,7 +38,7 @@ Prisma Client provides the following metrics:
   - `prisma_client_queries_wait`: The number of Prisma Client queries currently waiting for a connection because all connections are in use.
   - `prisma_pool_connections_busy`: The number of currently busy pool connections. These pool connections are currently executing a datasource query.
   - `prisma_pool_connections_idle`: The number of pool connections that are not currently being used. These pool connections are waiting for the next datasource query to run.
-  - `prisma_pool_connections_open`: The total number of [pool connections](/concepts/components/prisma-client/working-with-prismaclient/connection-pool#default-connection-pool-size) opened.
```
Comment (janpio) [inline content/200-concepts/100-components/02-prisma-client/240-metrics.mdx]: This is still not the number of "opened", but currently "open".

Doesn't this description come from the API when using it?

File: content/200-concepts/100-components/02-prisma-client/240-metrics.mdx [hunk 6]
```diff
@@ -38,7 +38,7 @@ Prisma Client provides the following metrics:
...
+  - `prisma_pool_connections_open`: The number of [pool connections](/concepts/components/prisma-client/working-with-prismaclient/connection-pool#default-connection-pool-size) opened.
```
Comment (janpio) [inline content/200-concepts/100-components/02-prisma-client/240-metrics.mdx]: This is still not the number of "opened", but currently "open".

Doesn't this description come from the API when using it?

Comment (ruheni) [reply content/200-concepts/100-components/02-prisma-client/240-metrics.mdx]: Got it.

Side note: There are duplicate properties available, which I'm not sure if it's intentional or not.

```
    {
      "key": "prisma_pool_connections_open",
      "labels": {
        
      },
      "value": 1,
      "description": "Number of currently open Pool Connections"
    },
    {
      "key": "prisma_pool_connections_open",
      "labels": {
        
      },
      "value": 0,
      "description": "Number of currently open Pool Connections (able to execute a datasource query)"
    }
```

**Review by SevInf — APPROVED** [review]

File: content/200-concepts/100-components/02-prisma-client/240-metrics.mdx (hunk 6 above)
Comment (janpio) [reply content/200-concepts/100-components/02-prisma-client/240-metrics.mdx]: Is that in the real output of the latest `@dev` version? Should not be, would be a bug that we need to fix.

File: content/200-concepts/100-components/02-prisma-client/240-metrics.mdx [hunk 7]
```diff
@@ -306,47 +312,47 @@ console.log(metrics)
...
-prisma_client_queries_wait_histogram_ms_bucket{le="10"} 3
-prisma_client_queries_wait_histogram_ms_bucket{le="50"} 3
-prisma_client_queries_wait_histogram_ms_bucket{le="100"} 3
-prisma_client_queries_wait_histogram_ms_bucket{le="500"} 3
-prisma_client_queries_wait_histogram_ms_bucket{le="1000"} 3
-prisma_client_queries_wait_histogram_ms_bucket{le="5000"} 3
-prisma_client_queries_wait_histogram_ms_bucket{le="50000"} 3
-prisma_client_queries_wait_histogram_ms_bucket{le="+Inf"} 3
-prisma_client_queries_wait_histogram_ms_sum 0.023208
-prisma_client_queries_wait_histogram_ms_count 3
```
Comment (janpio) [inline content/200-concepts/100-components/02-prisma-client/240-metrics.mdx]: This should still exist.
Especially in current `dev` version.

File: content/200-concepts/100-components/02-prisma-client/240-metrics.mdx (hunk 6 above)
Comment (ruheni) [reply content/200-concepts/100-components/02-prisma-client/240-metrics.mdx]: Added it to my TODO to create an issue for it once I get a reproduction after the release.

File: content/200-concepts/100-components/02-prisma-client/240-metrics.mdx [hunk 8]
```diff
@@ -306,47 +312,47 @@ console.log(metrics)
...
-
-# HELP prisma_client_queries_duration_histogram_ms
+# HELP prisma_client_queries_duration_histogram_ms Histogram of the duration of all executed Prisma Client queries in ms
```
Comment (ruheni) [inline content/200-concepts/100-components/02-prisma-client/240-metrics.mdx:430]:
```suggestion
# HELP prisma_client_queries_duration_histogram_ms Histogram of the duration of all executed Prisma Client queries in ms
# HELP prisma_client_queries_wait_histogram_ms The wait time for a worker to get a connection.
# TYPE prisma_client_queries_wait_histogram_ms histogram
prisma_client_queries_wait_histogram_ms_bucket{le="0"} 0
prisma_client_queries_wait_histogram_ms_bucket{le="1"} 3
prisma_client_queries_wait_histogram_ms_bucket{le="5"} 3
prisma_client_queries_wait_histogram_ms_bucket{le="10"} 3
prisma_client_queries_wait_histogram_ms_bucket{le="50"} 3
prisma_client_queries_wait_histogram_ms_bucket{le="100"} 3
prisma_client_queries_wait_histogram_ms_bucket{le="500"} 3
prisma_client_queries_wait_histogram_ms_bucket{le="1000"} 3
prisma_client_queries_wait_histogram_ms_bucket{le="5000"} 3
prisma_client_queries_wait_histogram_ms_bucket{le="50000"} 3
prisma_client_queries_wait_histogram_ms_bucket{le="+Inf"} 3
prisma_client_queries_wait_histogram_ms_sum 0.023208
prisma_client_queries_wait_histogram_ms_count 3
```

File: content/200-concepts/100-components/02-prisma-client/240-metrics.mdx [hunk 9]
```diff
@@ -353,6 +353,21 @@ prisma_pool_connections_open 1
...
+prisma_client_queries_wait_histogram_ms_bucket{le="10"} 3
+prisma_client_queries_wait_histogram_ms_bucket{le="50"} 3
+prisma_client_queries_wait_histogram_ms_bucket{le="100"} 3
+prisma_client_queries_wait_histogram_ms_bucket{le="500"} 3
+prisma_client_queries_wait_histogram_ms_bucket{le="1000"} 3
+prisma_client_queries_wait_histogram_ms_bucket{le="5000"} 3
+prisma_client_queries_wait_histogram_ms_bucket{le="50000"} 3
+prisma_client_queries_wait_histogram_ms_bucket{le="+Inf"} 3
+prisma_client_queries_wait_histogram_ms_sum 0.023208
+prisma_client_queries_wait_histogram_ms_count 3
```
Comment (janpio) [inline content/200-concepts/100-components/02-prisma-client/240-metrics.mdx]: I think that is the wrong location. The `HELP` line above belongs to the metrics below.

Is that really how executing the command returns it!?

File: content/200-concepts/100-components/02-prisma-client/240-metrics.mdx [hunk 10]
```diff
@@ -306,30 +312,47 @@ console.log(metrics)
...
+prisma_pool_connections_open 1
 
-# HELP prisma_client_queries_active
-# TYPE prisma_client_queries_active gauge
-prisma_client_queries_active 0
+# HELP prisma_pool_connections_open Number of currently open Pool Connections (able to execute a datasource query)
+# TYPE prisma_pool_connections_open gauge
+prisma_pool_connections_open 0
 
+# HELP prisma_client_queries_duration_histogram_ms Histogram of the duration of all executed Prisma Client queries in ms
```
Comment (janpio) [inline content/200-concepts/100-components/02-prisma-client/240-metrics.mdx]:
```suggestion
```

File: content/200-concepts/100-components/02-prisma-client/240-metrics.mdx [hunk 11]
```diff
@@ -345,8 +368,6 @@ prisma_client_queries_wait_histogram_ms_bucket{le="50000"} 3
 prisma_client_queries_wait_histogram_ms_bucket{le="+Inf"} 3
 prisma_client_queries_wait_histogram_ms_sum 0.023208
 prisma_client_queries_wait_histogram_ms_count 3
-
-# HELP prisma_client_queries_duration_histogram_ms
 # TYPE prisma_client_queries_duration_histogram_ms histogram
```
Comment (janpio) [inline content/200-concepts/100-components/02-prisma-client/240-metrics.mdx:431]:
```suggestion

# HELP prisma_client_queries_duration_histogram_ms Histogram of the duration of all executed Prisma Client queries in ms
# TYPE prisma_client_queries_duration_histogram_ms histogram
```

File: content/200-concepts/100-components/02-prisma-client/240-metrics.mdx (hunk 10 above)
Comment (janpio) [reply content/200-concepts/100-components/02-prisma-client/240-metrics.mdx]: plus https://github.com/prisma/docs/pull/5271/files#r1324162989

File: content/200-concepts/100-components/02-prisma-client/240-metrics.mdx (hunk 9 above)
Comment (ruheni) [reply content/200-concepts/100-components/02-prisma-client/240-metrics.mdx]: Yes.

**Pull request merged**